src/utils.ts

```diff
@@ -42,7 +43,7 @@ export function findHeadingBoundary(
 }
 
 
-export async function createDirNotExist(filepath: string) {
-  const directory = path.dirname(filepath)
-  if (!await this.app.vault.adapter.exists(directory)) await this.app.vault.adapter.mkdir(directory)
```

Comment:
@SettingDust -- I think we really should be using the Obsidian API instead of the `fs` module where possible, including here.  Although in most cases, we'll be operating on an actual filesystem, it's totally possible that Obsidian will in some cases abstract away something that _can't_ be accessed via the `fs` module.

Because I'm not sure it's clear -- what I'm fairly sure was happening before is that the `adapter.mkdir` method *does not* know how to handle recursively creating multiple directories, and was instead creating a single directory with the path separator (`/`) in its name.

To work around this limitation, though, is really easy -- all you need to do is split `directory` on the path separator, and in your code, create each necessary path segment, one at a time.

Reply:
I've tried. Obsidian always think the root of path is vault root. Can't cd to sub dir. So, create `a` then `a/b` still not working

Reply:
Can you not just create the directory structure from the top down?

ie 

```
await app.vault.createFolder("TestTopDir")
await app.vault.createFolder("TestTopDir/Inner")
```

Then do as @coddingtonbear suggested and loop through until you get to the base.

Reply:
Talked. Creating `a` then `a/b` won't work as expected